src/TreeSupport.cpp

```diff
@@ -596,18 +596,32 @@ void TreeSupport::generateContactPoints(const SliceMeshStorage& mesh, std::vecto
     }
 }
 
-void TreeSupport::insertDroppedNode(std::vector<Node*>& nodes_layer, Node* p_node)
+TreeSupport::Node* TreeSupport::insertDroppedNode(std::vector<Node*>& nodes_layer, Node* p_node)
 {
-    std::vector<Node*>::iterator conflicting_node_it = std::find(nodes_layer.begin(), nodes_layer.end(), p_node);
+    const std::function<bool(Node*)> node_ptr_eq{ [&p_node](Node* p_other) { return p_node->position == p_other->position; } };
+    std::vector<Node*>::iterator conflicting_node_it = std::find_if(nodes_layer.begin(), nodes_layer.end(), node_ptr_eq);
```

Comment:
```suggestion
    auto node_ptr_eq{ [&p_node](Node* p_other) { return p_node->position == p_other->position; } };
    auto conflicting_node_it = std::find_if(nodes_layer.begin(), nodes_layer.end(), node_ptr_eq);
```

Reply:
Yup. I started putting in the types when it kept failing, only to discover I didn't use `find_if` but had kept `find` instead. Then forgot to put back the `auto` when I fixed it.